File: 프로그래머스/Python/0단계/직각 삼각형 출력하기.py
# 정수 n 이 주어지면 높이와 너비가 n 인 직각 이등변 삼각형을 출력하도록 코드를 작성해보세요.
# 별 찍는 원리를 까먹어서 당황한 문제

# 내 풀이

n = int(input())
for i in range(n):
    for j in range(i+1):
        print('*', end = '')
    print()


# 다른사람 풀이

print('\n'.join('*' * (i + 1) for i in range(int(input()))))

File: 프로그래머스/JavaScript/0단계/직각삼각형 출력하기.js
// 별출력하는 문제
// readline모듈에 관해서 알게되었다.
// https://lakelouise.tistory.com/140


// 내 풀이
const readline = require('readline');  // 모듈 가져오기
const rl = readline.createInterface({  //readline 모듈을 이용해 입출력을 위한 인터페이스 객체 생성
    input: process.stdin,
    output: process.stdout
});

//rl 변수
rl.on('line', function (line) {  // 한 줄씩 입력받은 후 실행할 코드
    input = line;   // 입력된 값은 line에 저장된다. 
}).on('close', function () {   
    let InputNum = Number(input);  // 입력이 끝난 후 실행할 코드 
    for (let i = 1; i <= InputNum; i++){
        console.log('*'.repeat(i));
    }
});
